src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -357,6 +357,11 @@ internal ValueTask<ImmutableArray<Diagnostic>> GetSourceGeneratorDiagnosticsAsyn
             return _solution.CompilationState.GetSourceGeneratorDiagnosticsAsync(this.State, cancellationToken);
         }
 
+        internal ValueTask<GeneratorDriverRunResult?> GetSourceGeneratorRunResultAsync(CancellationToken cancellationToken)
```

Comment:
@jasonmalinowski In a previous version of this PR, we discussed if we should replace `GetSourceGeneratorDiagnosticsAsync` with this method. 

I chose to keep them separate because the diagnostics path actually returns data in the `GeneratedFileReplacingCompilationTracker` path, where the run result just throws. If we can figure out what the semantics of get run result would be in the `GeneratedFileReplacingCompilationTracker` then we could collapse them into a single method.